core/modules/layout/layout.module

```diff
@@ -1533,3 +1534,17 @@ function layout_set_handler_contexts(array &$contexts, array $all_contexts, arra
   // All required contexts populated successfully.
   return TRUE;
 }
+
+/**
+ * Implements hook_block_info().
+ */
+function layout_block_info() {
+  $blocks = array();
+  $blocks['text_block'] = array(
+    'info' => t('Text block'),
+    'description' => t('Create a text block.'),
+    'class' => 'TextBlock',
```

Comment:
Conferring with @jenlampton, terminology here could be changed:

- info: "Add custom block"
- description: "A basic block for adding custom text."
- class: BlockText

This way the dialog will show "Add custom block" (as you will be able to do once we integrate this with block.module), and the description "A basic block for adding custom text." will be shown in the admin preview.